### ISSUE TITLE
Compatibility with PHP 8.4

### ISSUE THREAD

**Issue by andypost** [phpspec/phpspec#1477]
Implicitly nullable arguments https://wiki.php.net/rfc/deprecate-implicitly-nullable-types

- fixed in 7.5 https://github.com/phpspec/phpspec/pull/1472 but needs to fix
- 8.x https://github.com/phpspec/phpspec/pull/1473

Deprecation of `E_STRICT` https://wiki.php.net/rfc/deprecations_php_8_4#remove_e_strict_error_level_and_deprecate_e_strict_constant

- 7.5 #1479 
- 8.0 #1478 

finally CI job https://github.com/phpspec/phpspec/pull/1476 and new release

**Comment by mpysiak** [comment]
ping! 😄

**Comment by JanPalen** [comment]
Is there any info when  PHP8.4 support will be added? That will be very helpful.

**Comment by JurRutten** [comment]
+1

**Comment by rimas-kudelis** [comment]
The main branch has had no new commits for 4 months now. PHP 8.4, as well as the support patches for it, have been around for three months, but are neither being merged in, nor discussed. Is phpspec being abandoned?

**Comment by torchello** [comment]
@ciaranmcnulty @MarcelloDuarte hey folks! could you please comment on the state of the project?

**Comment by NenadStef** [comment]
Hi friends

Is there any info about state of this project or release date of the new version that will have PHP8.4 support?

**Comment by MarcelloDuarte** [comment]
As per @ciaranmcnulty post below, he has been looking for a lead maintainer to take over. I am happy to act as a maintainer in the interim, but I need someone to catch me up with the project

![image](https://user-images.githubusercontent.com/144535/423497040-7bc367ca-0dd9-4c8f-9a54-8e6c4bfa0579.jpeg)

**Comment by jmleroux** [comment]
Hell @MarcelloDuarte , could you please point me to this post please?

_To have a bit more context about the state of this project_

> As per [@ciaranmcnulty](https://github.com/ciaranmcnulty) post below, he has been looking for a lead maintainer to take over. I am happy to act as a maintainer in the interim, but I need someone to catch me up with the project

**Comment by torchello** [comment]
@MarcelloDuarte Thanks! Any volunteers so far? I can potentially (co-)lead the project in a few months. Meanwhile, can we get it ready for PHP 8.4? 

Can you review the PRs from the issue description?

**Comment by MarcelloDuarte** [comment]
Hey @torchello, thanks for this. I had a look and #1473 and #1478 seem fine, the other heading for 7.5 looks good too. Is 7.5 supposed to support php 8.4 (I say this because of the red CI in #1477). Anyway, happy to merge 73 and 78. What else we need to support 8.4?

**Comment by torchello** [comment]
Seems like CI config has to be updated to include PHP 8.4. Let's merge https://github.com/phpspec/phpspec/pull/1473 and https://github.com/phpspec/phpspec/pull/1478. Then I will create a PR to add 8.4 and we'll see if CI is green. Wdyt?

**Comment by MarcelloDuarte** [comment]
@torchello both merged. Would be good to check other components (e.g. prophecy) to ensure they will be compatible with the new major (8.0) so we don't get stopped like this time around. Should be fine, I think.

**Comment by andypost** [comment]
Looks only CI needs work #1476  and 7.5 backport #1479

**Comment by torchello** [comment]
Sorry for delay, folks! I will have some time this week.

**Comment by JurRutten** [comment]
Bump

**Comment by andypost** [comment]
looks it needs follow-up for PHP 8.5 as [first alpha is out](https://wiki.php.net/todo/php85) and deprecations on a way https://wiki.php.net/rfc/deprecations_php_8_5

**Comment by torchello** [comment]
@andypost good point! I just pushed the (potential) fix for https://github.com/phpspec/phpspec/pull/1493 (can't reproduce it locally). We can start getting ready for 8.5 after finalizing the 8.4 release.
